bin/check-azurerm-monitor-metric.rb

```diff
@@ -0,0 +1,305 @@
+      unknown 'At least one threshold must be provided.'
+    end
+
+    if last_metric_values.empty?
+      unknown "There are no metric values for #{config[:metric]} on resource #{config[:resource_id] || config[:resource_name]} with aggregation #{config[:aggregation]}"
+    else
+      last_metric_values.each do |metric_val|
+        if config[:critical_over] && metric_val[:value] > config[:critical_over].to_f
+          critical "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:warning_over] && metric_val[:value] > config[:warning_over].to_f
+          warning "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:critical_under] && metric_val[:value] < config[:critical_under].to_f
+          critical "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        elsif config[:warning_under] && metric_val[:value] < config[:warning_under].to_f
+          warning "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        else
+          ok "Metric #{metric_val[:metric_name]} is #{metric_val[:value]}"
+        end
+      end
+    end
+  end
+
+  def last_metric_values
+    @last_metric_values ||= find_last_metric_values
+  end
+
+  def find_last_metric_values
+    values = []
+    metric_response[:value].each do |metric_resp_value|
+      name = metric_resp_value[:name] ? metric_resp_value[:name][:value] : ''
+
+      next if metric_resp_value[:timeseries].empty?
+
+      metric_resp_value[:timeseries].each do |ts|
+        ts[:data].reverse_each do |metric_value|
+          if metric_value[config[:aggregation].to_sym]
+            values << {
+              value: metric_value[config[:aggregation].to_sym].to_f,
+              metric_name: name
+            }
+
+            break
+          end
+        end
+      end
+    end
+
+    values
+  end
+
+  def metric_response
+    auth_header = if config[:use_assigned_identity]
+                    uri = URI.parse("http://localhost:#{config[:local_auth_port]}/oauth2/token?resource=https://management.azure.com/")
+
+                    res = Net::HTTP.start(uri.host, uri.port, use_ssl: false) do |http|
+                      req = Net::HTTP::Get.new(uri)
+                      req['Metadata'] = 'true'
+                      http.request(req)
+                    end
+
+                    handle_response(res)
+
+                    auth_resp = JSON.parse(res.body, symbolize_names: true)
+
+                    "#{auth_resp[:token_type]} #{auth_resp[:access_token]}"
+                  else
+                    provider = MsRestAzure::ApplicationTokenProvider.new(
+                      config[:tenant_id],
+                      config[:client_id],
+                      config[:client_secret]
+                    )
+
+                    provider.get_authentication_header
+                  end
+
+    begin
+      url = "https://management.azure.com#{resource}/providers/microsoft.insights/metrics?" \
```

Comment:
seeing as we have seen this url twice in the file we should at least variablize it if not making it a cli arg.